lib/network-exporters/utils/uploadZipToUploadThing.ts

```diff
@@ -0,0 +1,39 @@
+import { File } from 'node:buffer';
+import { readFileSync } from 'node:fs';
+import { unlink } from 'node:fs/promises';
+import { utapi } from '~/app/api/uploadthing/core';
+
+export const uploadZipToUploadThing = async (
+  zipLocation: string,
+  fileName: string,
+) => {
+  try {
+    const zipBuffer = readFileSync(zipLocation);
+    const zipFile = new File([zipBuffer], `${fileName}.zip`, {
+      type: 'application/zip',
+    });
+
+    const { data, error } = await utapi.uploadFiles(zipFile);
```

Comment:
Just check if there is any option to expire data sent to uploadThing automatically. This is a feature of S3, but they might not expose it. Would be cool if they did.

Reply:
Or is there any other way of automatically deleting this data after download?

Reply:
There's a method called `getSignedURL` which helps to create a URL with expiration time, but this function only provides the URL and is not able to delete the file from UploadThing, and also this function is not available unless the user allows overrides in his app settings on the UploadThing dashboard. Ref: https://docs.uploadthing.com/api-reference/server#getsignedurl

Reply:
Currently UploadThing doesn't provide any API to auto-delete the file, ref: https://github.com/pingdotgg/uploadthing/issues/127#issuecomment-1758161108

Reply:
The only way we could delete the zip file from UploadThing is to write a server-side logic (could be a tRPC route or a server action) that is triggered after the zip file has been downloaded from the client-side.